pkg/bus/bus_test.go

```diff
@@ -13,82 +13,118 @@ type testQuery struct {
 	Resp string
 }
 
-func TestDispatchCtxCanUseNormalHandlers(t *testing.T) {
+func TestDispatch(t *testing.T) {
 	bus := New()
 
-	handlerWithCtxCallCount := 0
-	handlerCallCount := 0
+	var invoked bool
 
-	handlerWithCtx := func(ctx context.Context, query *testQuery) error {
-		handlerWithCtxCallCount++
+	bus.AddHandler(func(query *testQuery) error {
+		invoked = true
 		return nil
-	}
+	})
 
-	handler := func(query *testQuery) error {
-		handlerCallCount++
-		return nil
-	}
+	err := bus.Dispatch(&testQuery{})
+	require.Nil(t, err)
```

Comment:
Maybe you can use `require.NoError` for checking the absence of errors? Doesn't make much difference, but should make eventual failures more explicit.